service/awsclusterconfig/v1/resource_set.go

```diff
@@ -0,0 +1,138 @@
+package v1
+
+import (
+	"context"
+
+	"github.com/cenkalti/backoff"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/giantswarm/operatorkit/framework"
+	"github.com/giantswarm/operatorkit/framework/resource/metricsresource"
+	"github.com/giantswarm/operatorkit/framework/resource/retryresource"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/giantswarm/cluster-operator/service/awsclusterconfig/v1/key"
+	"github.com/giantswarm/cluster-operator/service/awsclusterconfig/v1/resource/awsconfig"
+)
+
+const (
+	// ResourceRetries presents number of retries for failed Resource
+	// operation before giving up.
+	ResourceRetries uint64 = 3
+)
+
+// ResourceSetConfig contains necessary dependencies and settings for
+// AWSClusterConfig framework ResourceSet configuration.
+type ResourceSetConfig struct {
+	K8sClient kubernetes.Interface
+	Logger    micrologger.Logger
+
+	HandledVersionBundles []string
+	ProjectName           string
+}
+
+// NewResourceSet returns a configured AWSClusterConfig framework ResourceSet.
+func NewResourceSet(config ResourceSetConfig) (*framework.ResourceSet, error) {
+	var err error
+
+	if config.K8sClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.K8sClient must not be empty")
+	}
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.Logger must not be empty")
+	}
+	if config.ProjectName == "" {
+		return nil, microerror.Maskf(invalidConfigError, "config.ProjectName must not be empty")
+	}
+
+	var awsConfigResource framework.Resource
+	{
+		c := awsconfig.Config{
+			K8sClient: config.K8sClient,
+			Logger:    config.Logger,
+		}
+
+		ops, err := awsconfig.New(c)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+
+		awsConfigResource, err = toCRUDResource(config.Logger, ops)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	resources := []framework.Resource{
+		awsConfigResource,
+	}
+
+	// Wrap resources with retry and metrics.
+	{
+		retryWrapConfig := retryresource.WrapConfig{}
+
+		retryWrapConfig.BackOffFactory = func() backoff.BackOff { return backoff.WithMaxTries(backoff.NewExponentialBackOff(), ResourceRetries) }
+		retryWrapConfig.Logger = config.Logger
+
+		resources, err = retryresource.Wrap(resources, retryWrapConfig)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+
+		metricsWrapConfig := metricsresource.WrapConfig{}
+
+		metricsWrapConfig.Name = config.ProjectName
+
+		resources, err = metricsresource.Wrap(resources, metricsWrapConfig)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	initCtxFunc := func(ctx context.Context, obj interface{}) (context.Context, error) {
+		return ctx, nil
+	}
+
+	handlesFunc := func(obj interface{}) bool {
+		_, err := key.ToCustomObject(obj)
+		if err != nil {
+			return false
+		}
+
+		// Currently there is only one version to be handled. As long as the
+		// object is of right type, it's good to go.
+
+		return true
```

Comment:
The version bundle version check is now missing here. Lets please add this. It guarantees more reliable execution of dedicated resources in the long run.

Reply:
Good point. Version bundle version information was actually missing from related CRs: https://github.com/giantswarm/apiextensions/pull/65

Reply:
...and I got mixed here. The referenced PR doesn't actually have anything to do with this. I'll fix this piece of code and there's separate discussion in that PR.

Reply:
...or not. I think that referenced change is specifically needed so that this `handleFunc` implementation can determine provider specific `cluster-operator` CR version and then select correct resource implementation for it.